### PR TITLE
Feature/revsdl 2526 socket channel unit tests

### PR DESCRIPTION
**Pull request by bonewell**

### CODE REVIEW

File: src/components/vr_module/src/socket_channel.cc [hunk 1]
```diff
@@ -56,8 +60,10 @@ bool SocketChannel::Start() {
   LOG4CXX_AUTO_TRACE(logger_);
   std::string address = "127.0.0.1";
```
Comment (DKlimenko) [inline src/components/vr_module/src/socket_channel.cc]: Could you please move socket address and port to ini file

File: src/components/vr_module/test/src/socket_channel_test.cc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+namespace net {
+
+class SocketChannelTest : public ::testing::Test {
+ protected:
+};
+
+TEST_F(SocketChannelTest, Start) {
+  MockConnectedSocket* socket = new MockConnectedSocket();
```
Comment (DKlimenko) [inline src/components/vr_module/test/src/socket_channel_test.cc]: Please add delete for new calls

**Comment by DKlimenko** [comment]
Reviewed

File: src/components/vr_module/src/socket_channel.cc (hunk 1 above)
Comment (bonewell) [inline src/components/vr_module/src/socket_channel.cc]: Yes, I can.

File: src/components/vr_module/test/src/socket_channel_test.cc (hunk 2 above)
Comment (bonewell) [inline src/components/vr_module/test/src/socket_channel_test.cc]: I didn't get what you mean. SocketChannel delete socket.

File: src/components/vr_module/include/vr_module/socket_channel.h [hunk 3]
```diff
@@ -47,6 +47,7 @@ namespace vr_module {
 class SocketChannel : public Channel {
  public:
   SocketChannel();
+  explicit SocketChannel(net::ConnectedSocket *socket);
```
Comment (DKlimenko) [inline src/components/vr_module/include/vr_module/socket_channel.h]: Is it possible to move star to the type?

Comment (bonewell) [inline src/components/vr_module/include/vr_module/socket_channel.h]: Why? I have heard that it is good practice to use star near variable. Google style allows to use star that way.

Comment (DKlimenko) [inline src/components/vr_module/include/vr_module/socket_channel.h]: Google style alows both of them, but from my point of view star relates to type not to variable name

Comment (bonewell) [inline src/components/vr_module/include/vr_module/socket_channel.h]: [Is ``int* p;'' right or is ``int *p;'' right?](http://www.stroustrup.com/bs_faq2.html#whitespace)
So, I will use your version for C++ code from now.

**Pull request merged**